docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+
+[source,terminal]
+----
+labels:
+- "key1=val1"
+- "key2=val2"
+----
+
+The key of a label can be subdomain-prefixed:
+
+[source,terminal]
+----
+labels:
+- "konveyor.io/key1=val1"
+----
+
+The value of a label can be empty:
+
+[source,terminal]
+----
+labels:
+- "konveyor.io/key="
+----
+
+The value of a label can be omitted. In that case, it is treated as an empty value:
+
+[source,terminal]
+----
+labels:
+- "konveyor.io/key"
+----
+
+.Reserved labels
+
+The analyzer defines some labels that have special meaning as follows:
+
+* `konveyor.io/source`: Identifies the source technology to which a rule or a ruleset applies
+
+* `konveyor.io/target`: Identifies the target technology to which a rule or a ruleset applies
+
+.Label selector
+
+The analyzer CLI takes the `--label-selector` field as an option. It is a string expression that supports logical AND, OR and NOT operations. You can use it to filter-in or filter-out rules by their labels.
+
+_Examples:_
+
+* To filter-in all rules that have a label with the key `konveyor.io/source` and value `eap6`:
++
+`--label-selector="konveyor.io/source=eap6"`
+
+* To filter-in all rules that have a label with the key `konveyor.io/source` and any value:
++
+`--label-selector="konveyor.io/source"`
+
+* To perform logical AND operations on matches of multiple rules using the `&&` operator:
++
+`--label-selector="key1=val1 && key2"`
+
+* To perform logical OR operations on matches of multiple rules using the `||` operator:
++
+`--label-selector="key1=val1 || key2"`
+
+* To perform a NOT operation to filter-out rules that have `key1=val1` label set using the `!` operator:
++
+`--label-selector="!key1=val1"`
+
+* To group sub-expressions and control precedence using AND:
++
+`--label-selector="(key1=val1 || key2=val2) && !val3"`
+
+.Dependency labels
+
+The analyzer engine adds labels to dependencies. These labels provide additional information about a dependency, such as its programming language and whether the dependency is open-source or internal.
+
+Currently, the analyzer adds the following labels to dependencies:
+
+[source,terminal]
+----
+labels:
+- konveyor.io/dep-source=internal
+- konveyor.io/language=java
+----
+
+.Dependency label selector
+
+The analyzer CLI accepts the `--dep-label-selector` option, which allows filtering-in or filtering-out incidents generated from a dependency by their labels.
+
+For example, the analyzer adds a `konveyor.io/dep-source` label to dependencies with a value that indicates whether the dependency is a known open-source dependency.
+
+To exclude incidents for all such open-source dependencies, you can use `--dep-label-selector` as follows:
+
+`konveyor-analyzer ... --dep-label-selector !konveyor.io/dep-source=open-source`
+
+The Java provider in the analyzer can also add an exclude label to a list of packages. To exclude all such packages, you can use `--dep-label-selector` as follows:
+
```

Comment:
Suggestion: "...add an exclude label `!` to a list..." or is '!" as "exclude" obvious to our users? I'd to see something after "exclude label" but that might just be me. Up to you, completely. I'm guessing that "!" = "an exclude label." If that's wrong,or even questionable, please ignore.